src/main/java/world/bentobox/bentobox/util/teleport/SafeSpotTeleport.java

```diff
@@ -230,17 +225,54 @@ private void checkChunks(final List<ChunkSnapshot> chunkSnapshot) {
      * @return true if a safe spot was found
      */
     private boolean scanChunk(ChunkSnapshot chunk) {
-        // Run through the chunk
-        for (int x = 0; x< 16; x++) {
+        int startY = location.getBlockY();
+        int minY = location.getWorld().getMinHeight();
+        int maxY = 60; // Just a dummy value
+
+        // Check the safe spot at the current height
+        for (int x = 0; x < 16; x++) {
             for (int z = 0; z < 16; z++) {
-                // Work down from the entry point up
-                for (int y = Math.min(chunk.getHighestBlockYAt(x, z), maxHeight); y >= 0; y--) {
-                    if (checkBlock(chunk, x,y,z)) {
+                if (minY >= startY && checkBlock(chunk, x, startY, z)) {
+                    return true;
+                }
+                maxY = Math.max(chunk.getHighestBlockYAt(x, z), maxY);
+            }
+        }
+        maxY = Math.min(maxY, maxHeight);
+
+        // Expand the height up and down until a safe spot is found
+        int upperY = startY + 1;
+        int lowerY = startY - 1;
+        boolean checkUpper = upperY <= maxY;
+        boolean checkLower = lowerY >= minY;
+        int limitRange = plugin.getSettings().getSafeSpotSearchVerticalRange(); // Limit the y-coordinate range
```

Comment:
Maybe just use `location.getWorld().getMaxHeight()` to work out an appropriate limit?

Reply:
Explained